### ISSUE TITLE
SCA check ID 27028 uses wrong check conditions CIS Microsoft Windows Server 2022 Benchmark v1.0.0

### ISSUE THREAD

**Issue by oleKHard** [wazuh/wazuh#17943]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.* | SCA | Agent | cis_win2022.yml | Windows Server 2022 |

Check: Domain member: Maximum machine account password age' is set to '30 or fewer days, but not 0'.

Remediation
 To establish the recommended configuration via GP, set the following UI path to 30 or fewer days, but not 0: Computer Configuration\Policies\Windows Settings\Security Settings\Local Policies\Security Options\Domain member: Maximum machine account password age
 
 Checks (Condition: all)

        c:net.exe accounts -> n:Maximum password age (days):\s+(\d+) compare <= 30
        c:net.exe accounts -> n:Maximum password age (days):\s+(\d+) compare > 0
        
 This is wrong check - net.accounts Maximum password age describes user accounts, not machine accounts. Check should be:
 Computer\HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\Netlogon\Parameters\MaximumPasswordAge <= 30
 Computer\HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\Netlogon\Parameters\MaximumPasswordAge > 0

Ref. https://www.stigviewer.com/stig/windows_7/2014-04-02/finding/V-3373

**Comment by juliamagan** [comment]
Another user has reported this error on [Slack](https://wazuh.slack.com/archives/C0A933R8E/p1730998690141659). From the check ID, the user appears to have a version earlier than 4.9.0 and Windows Server 2022.

After doing some testing, I was able to replicate this error with 4.9.2 and Windows 11, where the check ID is now 26023.

**Comment by juliamagan** [comment]
After investigating for a while, I could see that the error comes from the fact that the parentheses are not being escaped correctly and the regex check fails, i.e. the condition should be:

```yml
rules:

- 'c:net.exe accounts -> n:Maximum password age \(days\):\s+(\d+) compare <= 30'
- 'c:net.exe accounts -> n:Maximum password age \(days\):\s+(\d+) compare > 0'
```

**Comment by oleKHard** [comment]
I suggest you try to edit the group policy

Computer Configuration\Policies\Windows Settings\Security Settings\Local Policies\Security Options\Domain member: Maximum machine account password age

and then observe output of `net.exe accounts` and the reg key value `HKLM\SYSTEM\CurrentControlSet\Services\NetLogon\Parameters\MaximumPasswordAge`

**Comment by IsExec** [comment]
# Issue Update

The check is not correct in the previous versions because `net.exe accounts` is not applicable as it has to do with the Domain member. The correct check has been implemented in Windows server 2022 and Windows 11 as stated below:

## Microsoft Windows Server 2022

The check was corrected in the SCA Policy for Microsoft Windows server 2022 in **v4.9.0** and with CIS Microsoft Windows Server 2022 Benchmark v2.0.0 as shown below:

<details>
<summary> Check ID: 27026 </summary>

```yml
2.3.6.5 (L1) Ensure 'Domain member: Maximum machine account password age' is set to '30 or fewer days, but not 0'. (Automated)

condition: all
    rules:
      - 'r:HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\Netlogon\Parameters -> MaximumPasswordAge'
      - 'r:HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\Netlogon\Parameters -> MaximumPasswordAge -> n:^(\d+) compare <= 30'
      - 'not r:HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\Netlogon\Parameters -> MaximumPasswordAge -> 0'
```
</details>

Link to Issue:  https://github.com/wazuh/wazuh/issues/17339

## Microsoft Windows 11 Enterprise

The check has been corrected in SCA Policy for Microsoft Windows 11 Enterprise for **v4.10.0** release using CIS Benchmark v3.0.0 for Microsoft Windows 11 Enterprise as shown below:


<details>
<summary> Check ID: 26020</summary>

```yml
2.3.6.5 (L1) Ensure 'Domain member: Maximum machine account password age' is set to '30 or fewer days, but not 0'.(Automated)

    condition: all
    rules:
      - 'r:HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\Netlogon\Parameters -> MaximumPasswordAge'
      - 'not r:HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\Netlogon\Parameters -> MaximumPasswordAge -> 0'
      - 'r:HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\Netlogon\Parameters -> MaximumPasswordAge -> n:^(\d+) compare <= 30'
```
</details>

Link to Issue:  https://github.com/wazuh/wazuh/issues/22195

**Comment by ooniagbi** [comment]
@juliamagan the error has been fixed in 4.10.0 according to the review by @IsExec. Please confirm and we will close this as not planned.

**Comment by juliamagan** [comment]
After reviewing the updated CIS, it makes sense to change the policy with the registries. I have also done some tests where it can be seen that it works as expected:

![image](https://github.com/user-attachments/assets/6cb08e34-320a-43fa-9538-67312c3a652a)
![image](https://github.com/user-attachments/assets/81494979-3ef0-47a9-883f-1fc911fdeaee)


![image](https://github.com/user-attachments/assets/39c5fc39-9a95-47c2-b292-e4c16f5897e0)
![image](https://github.com/user-attachments/assets/7c3a1279-6689-42bd-94b8-db26c84d1bf2)
